x-pack/plugins/lens/public/indexpattern_datasource/format_column.test.ts

```diff
@@ -0,0 +1,163 @@
+    const result = fn(datatable, { columnId: 'test', format: 'number', decimals: 0 });
+    expect(result.columns[0].meta.params).toEqual({
+      id: 'number',
+      params: {
+        pattern: '0,0',
+      },
+    });
+  });
+
+  describe('parent format', () => {
+    it('set parent format with params', () => {
+      const result = fn(datatable, {
+        columnId: 'test',
+        format: '',
+        parentFormat: JSON.stringify({ id: 'wrapper', params: { wrapperParam: 123 } }),
+      });
+      expect(result.columns[0].meta.params).toEqual({
+        id: 'wrapper',
+        params: {
+          wrapperParam: 123,
+          id: 'number',
+        },
+      });
+    });
+
+    it('retain inner formatter params', () => {
+      datatable.columns[0].meta.params = { id: 'myformatter', params: { innerParam: 456 } };
+      const result = fn(datatable, {
+        columnId: 'test',
+        format: '',
+        parentFormat: JSON.stringify({ id: 'wrapper', params: { wrapperParam: 123 } }),
+      });
+      expect(result.columns[0].meta.params).toEqual({
+        id: 'wrapper',
+        params: {
+          wrapperParam: 123,
+          id: 'myformatter',
+          params: {
+            innerParam: 456,
+          },
+        },
+      });
+    });
+
+    it('overwrite existing wrapper param', () => {
+      datatable.columns[0].meta.params = {
+        id: 'wrapper',
+        params: { wrapperParam: 0, id: 'myformatter', params: { innerParam: 456 } },
+      };
+      const result = fn(datatable, {
+        columnId: 'test',
+        format: '',
+        parentFormat: JSON.stringify({ id: 'wrapper', params: { wrapperParam: 123 } }),
+      });
+      expect(result.columns[0].meta.params).toEqual({
+        id: 'wrapper',
+        params: {
+          wrapperParam: 123,
+          id: 'myformatter',
+          params: {
+            innerParam: 456,
+          },
+        },
+      });
+    });
+
+    it('overwrites format with well known pattern including decimals', () => {
+      datatable.columns[0].meta.params = {
+        id: 'previousWrapper',
+        params: { id: 'myformatter', params: { innerParam: 456 } },
+      };
+      const result = fn(datatable, {
+        columnId: 'test',
+        format: 'number',
+        decimals: 5,
+        parentFormat: JSON.stringify({ id: 'wrapper', params: { wrapperParam: 123 } }),
+      });
+      expect(result.columns[0].meta.params).toEqual({
+        id: 'wrapper',
+        params: {
+          wrapperParam: 123,
+          id: 'number',
+          params: {
+            pattern: '0,0.00000',
+          },
+        },
+      });
+    });
+  });
+
+  it('does not touch other column meta data', () => {
+    const extraColumn: DatatableColumn = { id: 'test2', name: 'test2', meta: { type: 'number' } };
+    datatable.columns.push(extraColumn);
+    const result = fn(datatable, { columnId: 'test', format: 'number' });
+    expect(result.columns[1]).toEqual(extraColumn);
+  });
+});
```

Comment:
Maybe add one more case for `parentFormat` without `id` field => in this case the `parentFormat` should be ignored.

Apart from that I cannot see any other left case